packages/cardano-services/src/StakePool/DbSyncStakePoolProvider/DbSyncStakePoolProvider.ts

```diff
@@ -199,10 +199,18 @@ export class DbSyncStakePoolProvider extends DbSyncProvider(RunnableModule) impl
     const poolUpdates = await this.#cache.get(queryCacheKey(StakePoolsSubQuery.POOL_HASHES, options), () =>
       this.#builder.queryPoolHashes(query, params)
     );
-    // Get total amount of ada cached
-    const totalAdaAmount = await this.#cache.get(queryCacheKey(StakePoolsSubQuery.TOTAL_ADA_AMOUNT), () =>
-      this.#builder.getTotalAmountOfAda()
-    );
+    // Get cached total staked amount
+    const totalStake = await this.#cache.get(queryCacheKey(StakePoolsSubQuery.TOTAL_STAKE), async () => {
+      const distribution = await this.cardanoNode.stakeDistribution();
+
+      for (const [_, value] of distribution) return value.stake.supply.toString();
```

Comment:
```suggestion
      const first = [...distribution.values()][0];
      if (!first) throw ...
      return first.stake.supply.toString();
```
Slightly improved semantics?

Reply:
Yes, I focused on the optimization:

- mine version access the `Map` through its built in `Iterator`, only once
- proposed version requires the creation of an `Array` from a `Map`